modules/matterhorn-scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java

```diff
@@ -1232,6 +1241,52 @@ public int compare(ARecord o1, ARecord o2) {
     return Stream.mk(result).bind(recordToMp).toList();
   }
 
+  /*
+   * Returns a list of conflicting mediapackages.  This method checks for containment, starts-during, and ends-during.
+   */
+  private List<MediaPackage> checkScheduleConflicts(Date checkStart, Date checkEnd, ARecord[] records) {
+    final List<ARecord> result = new ArrayList<>();
+    for (final ARecord r : records) {
+      final Date start = r.getProperties().apply(Properties.getDate(START_DATE_CONFIG));
+      final Date end = r.getProperties().apply(Properties.getDate(END_DATE_CONFIG));
+      /*
+      If the potential event starts during event r OR
+      If the potential event ends during event r OR
+      If the potential event begins before, and ends after event r (ie, containment) OR
+      If the potential event begins or ends within the minimum separation distance of event r
+      */
+      if (checkStart.after(start) && checkStart.before(end)
+       || checkEnd.after(start) && checkEnd.before(end)
+       || checkStart.before(start) && checkEnd.after(end)
+       || eventWithinMinimumSeparation(checkStart, checkEnd, start, end)) {
+        result.add(r);
+      }
+    }
+    result.sort(new Comparator<ARecord>() {
+      @Override
+      public int compare(ARecord o1, ARecord o2) {
+        Date start1 = o1.getProperties().apply(Properties.getDate(START_DATE_CONFIG));
+        Date start2 = o2.getProperties().apply(Properties.getDate(START_DATE_CONFIG));
+        return start1.compareTo(start2);
+      }
+    });
+    return Stream.mk(result).bind(recordToMp).toList();
+  }
+
+  /**
+   * Returns true of checkStart is within EVENT_MINIMUM_SEPARATION_SECONDS of either the start or end dates, or checkEnd
+   * is within EVENT_MINIMUM_SEPARATION_SECONDS of either the start or end dates.  False otherwise
+   */
+  private boolean eventWithinMinimumSeparation(Date checkStart, Date checkEnd, Date start, Date end) {
+    if (Math.abs(checkStart.getTime() - start.getTime()) <= EVENT_MINIMUM_SEPARATION_MILLISECONDS
```

Comment:
@gregorydlogan One last thing: Please change the '<=' in this tests to '<' for the following reason:

The Admin UI allows scheduling start, end and duration by minutes. If a user now wants Recording 1 (3:00PM-3:29PM, duration = 29 minutes) and Recording 2 (3:30PM-4:00PM), this will be reported as a conflict which is unclear to the user as from the data he sees and knows, there is no conflict here

If you change the "<=" to "<", that won't happen which improves usability quite a lot. Probably requires to adjust some comments.

Also, 3:29 to 3:30 has 60 seconds difference with means EVENT_MINIMUM_SEPARATION_SECONDS should be ok.